File: php/mqtt_connection.php
<?php
// PHP MQTT Connection Example

// Define some constants.
// Change username and password to values in Your Credentials.
$_USERNAME = '';
$_TOKEN = '';
$_TOKEN_HASH = md5($_TOKEN);
$_HOST = 'q.thingfabric.com';
$_PORT = 1883;		// Use port 8883 for SSL
$_TOPIC = '[domain]/test-stuff/test-thing';
$_PAYLOAD = '{"Hello":"World!"}';
$_QOS = 0;

// Using the PHP Mosquitto extension (https://github.com/mgdm/Mosquitto-PHP).
$client = new Mosquitto\Client();

// Bind callbacks
$client->onConnect('on_connect');
$client->onDisconnect('on_disconnect');
$client->onSubscribe('on_subscribe');
$client->onMessage('on_message');

// Set client credentials.
$client->setCredentials($_USERNAME, $_TOKEN_HASH);

$client->connect($_HOST, $_PORT);

// Subscribe to a topic.
$client->subscribe($_TOPIC, $_QOS);



// Loop forever to maintain a connection with the host.
for(;;){
	$client->loop();
}

// Define callback functions.
function on_connect($rc, $message) {
	global $_TOPIC, $_PAYLOAD;

	// Display the connection's result code and explanation message.
	echo 'Code: ' . $rc . ' (' . $message . ')' . PHP_EOL;

	// Publish to a topic.
	$client->publish($_TOPIC, $_PAYLOAD);
}
function on_disconnect() {
	global $_HOST, $_PORT, $_USERNAME;

	// Display notice/confirmation of being disconnected.
	echo 'Client disconnected.' . PHP_EOL;
	echo 'Will attempt to reconnect in 5s.' . PHP_EOL;

	sleep(5);

	// Create a new $client object.
	echo 'Trying to reconnect...' . PHP_EOL;

	$client = new Mosquitto\Client();

	$client->connect($_HOST, $_PORT);

	// Bind callbacks
	$client->onConnect('on_connect');
	$client->onDisconnect('on_disconnect');
	$client->onSubscribe('on_subscribe');
	$client->onMessage('on_message');

	for(;;){
		$client->loop();
	}
}
function on_subscribe() {
	// Confirm that the subscription has been made.
	echo 'Subscription confirmed.' . PHP_EOL;
}
function on_message($message) {
	// Display the received message's topic and payload.
	echo 'Message received on ' . $message->topic . ': ' . $message->payload . PHP_EOL;
}

?>

File: ruby/mqtt_connection.rb
# Ruby MQTT Connection Example

require 'rubygems'
require 'mqtt' 		# Using the mqtt gem
require 'digest'

# Define some constants.
# Change username and password to values found in Your Credentials.
USERNAME = ''
TOKEN = ''
TOKEN_HASH = Digest::MD5.hexdigest(TOKEN)
HOST = 'q.thingfabric.com'
PORT = 1883		# Use port 8883 for SSL.
TOPIC = '[domain]/test-stuff/test-thing'
PAYLOAD = '{"Hello":"World!"}'

# Connect to q.thingfabric.com using the following parameters.
MQTT::Client.connect(
	:client_id => "ruby-mqtt",
	:remote_host => HOST,
	:remote_port => PORT,
	:username => USERNAME,
	:password => TOKEN_HASH
	) do |client|
	# Confirm that the connection has been made.
	puts 'Connected to host.'

	# The publishing has to be done in a separate thread, as 
	# the subscription service locks up the current thread.
	Thread.new do
		# Publish a payload to a topic.
		client.publish(TOPIC, PAYLOAD)

		# Display a notice that the message was published.
		puts 'Published message.'
	end

	# This both subscribes to a topic and defines the callback function.
	client.get(TOPIC) do |topic, message|
		# Display the received message's topic and payload.
		puts "Received message on #{topic}: #{message}"
	end
end
